### ISSUE TITLE
Monaco hc themes do not follow Windows High Contrast theme colors

### ISSUE THREAD

**Issue by jayspadie** [Microsoft/monaco-editor#472]
monaco-editor version: 0.8.3
Browser: Edge
OS: Windows 10

During Accessibility testing of our product, we’ve gotten a few bugs that assert that Monaco should follow the Windows High Contrast themes when those are selected rather than overriding it with colors for language syntax highlighting and diffs, as is currently done by the hc-black Monaco theme.

I understand that there are technical reasons for supporting the rich user interface that requires setting -ms-high-contrast-adjust to none so that the simplified Windows High Contrast styling can be overridden, but there are also visual design decisions such as maintaining colored syntax highlighting rather than using only white text on a black background (or black on white) as specified by the Windows HC themes.

I understand that the current styling is "As Designed" and I personally don't disagree with it.  I would hate to lose syntax highlighting myself, but I can see the point that we're not following the intent of simplified, high contrast black/white coloring for those visually impaired users who may need it.

Should you decide to try to follow the minimal Windows High Contrast theme colors (for Edge/IE), there are a few resources out there where this is discussed including the [limited support for it](https://www.paciellogroup.com/blog/2016/12/windows-high-contrast-mode-the-limited-utility-of-ms-high-contrast/), a more positive [how to use it (including system colors that give the selected HC theme colors)](http://www.gwhitworth.com/blog/2017/04/how-to-use-ms-high-contrast), and a [JSFiddle that shows all the system color styles](https://jsfiddle.net/dwilson/gpL78gzy/1/).  For example, text color is _windowText_, selected text has a _highlight_ background, and the background color is _window_.

If everything in its current state is by design and won't change, then we may at least workaround the syntax highlighting question by creating our own theme(s) that defines everything as black or white (while still providing other language features - not just plain text).  But in any case, it would be good to make any official plans for Monaco publicly visible here for anyone else with the same question.  Thanks.

**Comment by alexandrudima** [comment]
Here is Visual Studio Community 2015 for reference:

![image](https://user-images.githubusercontent.com/5047891/27438610-1e2c7db4-5766-11e7-9a12-678c88dd9c2c.png)

![image](https://user-images.githubusercontent.com/5047891/27438631-33f946b8-5766-11e7-9f39-578aebc1a4b5.png)

**Comment by alexandrudima** [comment]
Thank you @jayspadie for the pointers to articles. Here is the editor now on master with the `vs` theme under Windows High Contrast Black for example:

![image](https://user-images.githubusercontent.com/5047891/27442141-b85343a6-576f-11e7-8bc1-434f5a8abc03.png)

![image](https://user-images.githubusercontent.com/5047891/27442208-db8f3a0a-576f-11e7-9806-7d9b4f74df4b.png)

So going forward there will be two (complementary) options for integrators:
1. do nothing, keep the editor on the `vs` theme, and on Edge we will now honour it's high contrast settings.
2. detect when the browser goes into HC mode (or have a user setting) and use `hc-black` as the editor theme. This should be used for e.g. Chrome

**Comment by jayspadie** [comment]
Sounds/Looks great!  Thanks @alexandrudima .  Nicely done.
